scapy/layers/ipsec.py

```diff
@@ -165,6 +165,270 @@ def data_for_encryption(self):
         return raw(self.data) + self.padding + struct.pack("BB", self.padlen, self.nh)  # noqa: E501
 
 
+###############################################################################
+if conf.crypto_valid_advanced:
+    from cryptography.exceptions import InvalidTag
+    from cryptography.hazmat.backends import default_backend
+    from cryptography.hazmat.primitives.ciphers import (
+        aead,
+    )
+else:
+    log_loading.info("Can't import python-cryptography v1.7+. "
+                     "Disabled IPsec encryption/authentication.")
+    default_backend = None
+    InvalidTag = Exception
+    aead = None
+
+###############################################################################
+
+
+class AEAD_Algo(object):
+    """
+    IPsec AEAD algorithm
+    """
+
+    def __init__(self, name, algo, block_size=None, iv_size=None,
+                 key_size=None, icv_size=None, salt_size=None):  # noqa: E501
+        """
+        :param name: the name of this encryption algorithm
+        :param algo: a Cipher module
+        :param block_size: the length a block for this algo. Defaults to the
+                           `block_size` of the cipher.
+        :param iv_size: the length of the initialization vector of this algo.
+                        Defaults to the `block_size` of the cipher.
+        :param key_size: an integer or list/tuple of integers. If specified,
+                         force the secret keys length to one of the values.
+                         Defaults to the `key_size` of the cipher.
+        :param icv_size: the length of the Integrity Check Value of this algo.
+                         Used by Combined Mode Algorithms e.g. GCM
+        :param salt_size: the length of the salt to use as the IV prefix.
+                          Usually used by Counter modes e.g. CTR
+        :param format_mode_iv: function to format the Initialization Vector
+                               e.g. handle the salt value
+                               Default is the random buffer from `generate_iv`
+        """
+        self.name = name
+        self.algo = algo
+        self.icv_size = icv_size
+
+        if block_size is not None:
+            self.block_size = block_size
+        else:
+            self.block_size = 1
```

Comment:
It's more pythony to set a default keyword if possible. Same below

Reply:
Done.
Same question as above: Do we need to do that for Crypt algo and Auth algo? (same logic...).

Reply:
I apply the same coding method as Crypt and Auth classes. Do I need to modify them aswell?